packages/code/src/renderingPlugin.tsx

```diff
@@ -228,6 +244,10 @@ export function flattenNestedTree(
       ];
     }
 
+    if (isTokenObject(val)) {
+      return [...acc, ...flattenNestedTree(val)];
+    }
+
     return acc;
```

Comment:
We might consider replacing this return acc with `enforceExhaustive(val)` to make sure we're handling all possible cases of val (which I think is what we want to do)

Reply:
We're more likely to not be handling specific data structures that we're unaware of than generic types. This PR is a good example of that.

Reply:
If you test enforceExhaustive(val) here, it seems that we've caught all the possible values. if you delete lines 247-249, enforceExhaustive will error -- which is why I'm suggesting

Reply:
The problem here is that it assumes we have our typing right for what val is (which, though I'm more confident after this PR that we do, but it's still possible). If you look at our main branch and do the same thing, it should also pass.

Reply:
Got it

Reply:
Since I don't have the same level of confidence here in the type correctness, I feel strongly we still need to `return acc` here to prevent this code from throwing at runtime. If we add enforceExhaustive, it marks `return acc` as unreachable code, which may not be true, and increases the chance someone removes that code in the future.